utils/check_content_style.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# check_content_style.py
+# Copyright (c) 2022 by tibetiroka
+#
+# Endless Sky is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <https://www.gnu.org/licenses/>.
+
+import regex as re
+import glob
+
+# Script that checks for common text content formatting pitfalls not covered by spellcheck or other tests.
+
+# Regex-based style checks for check_regexes()
+regexes = {re.compile(regex): description for regex, description in {
+	# Matches any space that is preceded only by tabs or '"' or '`' characters,
+	# except if it immediately follows a '"' or '`' character, as used in 'word'.
+	"^[\\t`\"]*(?<![`\"]) ": "indentations should use tabs only",
+	# Matches multiple space characters, unless they are followed by a '#', or if the text is a tooltip.
+	"(?<!^tip .*)(  $|  [^ #])": "multiple consecutive space characters",
+	# Matches any whitespace character at the end of the string.
+	"[^\\s]\\s+$": "trailing whitespace character",
+	# Matches any backticks at the end of the line that is
+	# preceded by the ending of a sentence ('?', '!', '.', ';', ':', ')', '"', ''', '>') and a whitespace.
+	# The same check fails if used on quotation marks, due to the irregular nature of 'word'.
+	"[?!.;:\\)\"'>] `$": "unnecessary space before closing backtick"
+}.items()}
```

Comment:
You can forgo the construction of an inner dict by iterating a list of 2-element lists